Development/nmos-cpp-node/node_implementation.cpp

```diff
@@ -72,6 +72,23 @@ namespace impl
 
         // smpte2022_7: controls whether senders and receivers have one leg (false) or two legs (true, default)
         const web::json::field_as_bool_or smpte2022_7{ U("smpte2022_7"), true };
+
+        // private_key_file: full path of private key file in PEM format
+        const web::json::field_as_string_or private_key_file{ U("private_key_file"), U("") };
+
+        // certificate_chain_file: full path of server certificate chain file in PEM format, which must be sorted
+        // starting with the server's certificate, followed by any intermediate CA certificates, and ending with the highest level (root) CA
+        const web::json::field_as_string_or certificate_chain_file{ U("certificate_chain_file"), U("") };
+
+        // rsa: full path of RSA private key file in PEM format and full path of server certificate chain file in PEM format, which must be sorted
+        // The RSA object like { "private_key_file": "server-rsa-key.pem", "certificate_chain_file": "server-rsa-chain.pem"}
+        // See private_key_file and certificate_chain_file above
+        const web::json::field_as_value rsa{ U("rsa") };
+
+        // ecdsa: full path of ECDSA private key file in PEM format and full path of server certificate chain file in PEM format, which must be sorted
+        // The ECDSA object like { "private_key_file": "server-ecdsa-key.pem, "certificate_chain_file": "server-ecdsa-chain.pem"}
+        // See private_key_file and certificate_chain_file above
+        const web::json::field_as_value ecdsa{ U("ecdsa") };
```

Comment:
> Since `rsa` and `ecdsa` are both optional, but if you include one of these objects, you need to supply both the `private_key_file` and the `certificate_chain_file`, I wonder why the latter two fields have defaults (empty string), but the object fields don't. If anything I would have expected it to be the other way round. At the moment, the cert loading functions will blithely attempt to open files with no names if these attributes are missing, wont they?

How about changing rsa and ecdsa to have a default object, with private_key_file and the certificate_chain_file two fields have the defaults (empty string), which would match the deprecated private_key_files and the certificate_chain_files (settings.h) style.

```
        // rsa: full paths of RSA private key file and server certificate chain file
        // the value must be an object like { "private_key_file": "server-rsa-key.pem", "certificate_chain_file": "server-rsa-chain.pem"}
        // see private_key_file and certificate_chain_file above
        const web::json::field_as_value_or rsa{ U("rsa"), web::json::value_of({ { private_key_file, U("") }, { certificate_chain_file, U("") } }) };

        // ecdsa: full paths of ECDSA private key file and server certificate chain file
        // the value must be an object like { "private_key_file": "server-ecdsa-key.pem, "certificate_chain_file": "server-ecdsa-chain.pem"}
        // see private_key_file and certificate_chain_file above
        const web::json::field_as_value_or rsa{ U("ecdsa"), web::json::value_of({ { private_key_file, U("") }, { certificate_chain_file, U("") } }) };
```

If the user wants to use the deprecated way to load the certificate, they should not register the new load_xxx_handler.

> By the way, does explicitly specifying specific `rsa` and `ecdsa` objects still make sense when considering future TLS 1.3 deployments?

The reason why we need the `rsa` and `ecdsa` objects is to support the EST implementation.